packages/ui/src/views/chatmessage/ChatMessage.jsx

```diff
@@ -720,6 +752,35 @@ export const ChatMessage = ({ open, chatflowid, isDialog, previews, setPreviews
         }
     }
 
+    const handleLeadCaptureSubmit = async (event) => {
+        if (event) event.preventDefault()
+        setIsLeadSaving(true)
+
+        const body = {
+            chatflowid,
+            chatId,
+            name: leadName,
+            email: leadEmail,
+            phone: leadPhone
+        }
+
+        const result = await leadsApi.addLead(body)
+        if (result.data) {
+            const data = result.data
+            if (!chatId) setChatId(data.chatId)
+            setLocalStorageChatflow(chatflowid, data.chatId, { lead: { name: leadName, email: leadEmail, phone: leadPhone } })
```

Comment:
here I got rid of the original implementation of saving as `<chatflowid>_LEAD`, to avoid saving more to local storage.

instead we store it to the `<chatflowid>_INTERNAL` since we already have it